src/swarm/neo/node/RequestSet.d

```diff
@@ -217,22 +225,49 @@ class RequestSet
 
         /***********************************************************************
 
-            Resumes the fiber or starts it if it is new or terminated.
+            Resumes the fiber or starts it if it is new or terminated. If
+            resuming (not starting) the fiber, `msg` is passed to the waiting
+            `suspendFiber()` call.
+
+            This method is called when a message for this request arrives. The
+            base class implements the basic behaviour and resumes the fiber.
+            However, in the node there is a case that needs special handing: the
+            message that starts the request.
+            When this initial message is received then the new `Request` object,
+            a subclass of `RequestOnConnBase`, is either created or fetched from
+            the object pool, and its `setReceivedPayload` is called, which calls
+            `resumeFiber_`. Since this is a new request the fiber needs to be
+            started rather than resumed. During the lifetime of the request
+            super class behaviour applies i.e. the fiber is resumed.
+
+            Params:
+                msg = the fiber message
 
         ***********************************************************************/
 
         override protected void resumeFiber_ ( fiber.Message msg )
         {
+            /*
+             * Detect whether the fiber should be started or resumed:
+             * If the fiber is in TERM state we obviously need to start it.
+             * The HOLD state is unfortunately ambiguous: A newly created fiber
+             * is in HOLD state just like a fiber that has been started and is
+             * suspended. With the fiber_was_started flag we distinguish between
+             * these cases.
+             */
+
             /* final */ switch (this.fiber.state)
             {
                 case fiber.state.HOLD:
                     if (this.fiber_was_started)
                     {
+                        // Fiber is suspended: Call super to resume the fiber.
                         super.resumeFiber_(msg);
                         break;
                     }
                     else
                     {
+                        // The fiber is new: Fall through to start it.
```

Comment:
I don't get this. The default case just asserts... doesn't look like it starts the fiber...?

Reply:
No, it goes to the next case: `case fiber.state.TERM: this.fiber.start();`.

Reply:
Ahh ok. Showing my lack of D2 knowledge ;)